components/server/src/authorization/spicedb.ts

```diff
@@ -40,63 +34,48 @@ export function spiceDBConfigFromEnv(): SpiceDBClientConfig | undefined {
     };
 }
 
-function spicedbClientFromConfig(config: SpiceDBClientConfig): Client {
-    const clientOptions: grpc.ClientOptions = {
-        "grpc.client_idle_timeout_ms": 10000, // this ensures a connection is not stuck in the "READY" state for too long. Required for the reconnect logic below.
-        "grpc.max_reconnect_backoff_ms": 5000, // default: 12000
-    };
-
-    return v1.NewClient(
-        config.token,
-        config.address,
-        v1.ClientSecurity.INSECURE_PLAINTEXT_CREDENTIALS,
-        undefined,
-        clientOptions,
-    ) as Client;
-}
-
-export class CachingSpiceDBClientProvider implements SpiceDBClientProvider {
+export class SpiceDBClientProvider {
     private client: Client | undefined;
 
-    private readonly interceptors: grpc.Interceptor[] = [];
-
     constructor(
-        private readonly _clientConfig: SpiceDBClientConfig,
-        private readonly clientCallMetrics?: IClientCallMetrics | undefined,
-    ) {
-        if (this.clientCallMetrics) {
-            this.interceptors.push(createClientCallMetricsInterceptor(this.clientCallMetrics));
-        }
-    }
+        private readonly clientConfig: SpiceDBClientConfig,
+        private readonly interceptors: grpc.Interceptor[] = [],
+    ) {}
 
     getClient(): SpiceDBClient {
-        let client = this.client;
-        if (!client) {
-            client = spicedbClientFromConfig(this.clientConfig);
-        } else if (client.getChannel().getConnectivityState(true) !== grpc.connectivityState.READY) {
-            // (gpl): We need this check and explicit re-connect because we observe a ~120s connection timeout without it.
-            // It's not entirely clear where that timeout comes from, but likely from the underlying transport, that is not exposed by grpc/grpc-js
-            client.close();
-
-            log.warn("[spicedb] Lost connection to SpiceDB - reconnecting...");
-
-            client = spicedbClientFromConfig(this.clientConfig);
-        }
-        this.client = client;
-
-        return client.promises;
-    }
-
-    protected get clientConfig() {
-        const config = this._clientConfig;
-        if (this.interceptors) {
-            return {
-                ...config,
-                options: {
-                    interceptors: [...this.interceptors],
+        if (!this.client) {
+            this.client = v1.NewClient(
+                this.clientConfig.token,
+                this.clientConfig.address,
+                v1.ClientSecurity.INSECURE_PLAINTEXT_CREDENTIALS,
+                undefined, //
+                {
+                    // we ping frequently to check if the connection is still alive
+                    "grpc.keepalive_time_ms": 1000,
```

Comment:
What about setting this up to ~4000 to avoid the "too many pings" responses?

Reply:
I haven't seen a too many pings response. have you?

Reply:
Yes: `message":"E Connection to dns:spicedb.default.svc.cluster.local:50051 at 10.43.248.133:50051 rejected by server because of excess pings. Increasing ping interval to 2000 ms","loggedViaConsole":true}`

It was on my own branch, but with these keepalive settings.

Reply:
Since it self-adjusts the ping interval. It would be nice to deploy this setting and see where we end up? Maybe 2000 is enough maybe 4000 is not. WDYT?

Reply:
>  It would be nice to deploy this setting and see where we end up?

I fear we'll never revisit and notice this but just accept that it's slower than necessary, especially when there a are (way) more requests than in preview envs.
How would we notice this in prod?

Reply:
This is just the answer to the pings and has not much to do with other request I assume.
I thought we would see the logs, but I guess they are traced, right?

Reply:
Why would it be slower?

Reply:
> Why would it be slower?

The server sends a GOAWAY (that translates to the log message above), which - depending on the server config - leads to turning down all other connection attempts for some time.

Reply:
Are you sure the server does not except connections after GOAWAY? How do you know?

Reply:
> Are you sure the server does not except connections after GOAWAY?

It matches by observation from looking of a lot of gRPC traces. I can't be :100: certain, though, because it can also just be coincidence. :upside_down_face:

Reply:
I think it will just not accept any request on that particular connection. So the client will reconnect with a longer ping interval until it again gets the GOAWAY.

Reply:
:shrug: